src/pages-and-resources/calculator/Settings.jsx

```diff
@@ -0,0 +1,46 @@
+import { getConfig } from '@edx/frontend-platform';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import PropTypes from 'prop-types';
+import React from 'react';
+import AppSettingsModal, { useAppSettingsForm } from '../app-settings-modal/AppSettingsModal';
+
+import messages from './messages';
+
+const CALCULATOR_HELP_URL = getConfig().CALCULATOR_HELP_URL
+  || 'https://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/exercises_tools/calculator.html';
+
+function CalculatorSettings({
+  intl,
+  onClose,
+}) {
+  const {
+    values,
+    handleSubmit,
+    handleChange,
+    handleBlur,
+  } = useAppSettingsForm({
+    appid: 'calculator',
+  });
```

Comment:
when we will use `<Formik />` component with AppSettingsModal this will not be needed and we can manage stater right from `AppSettingsModal`.

Reply:
We cannot entirely manage the state from there since it will not know the exact shape of the data, only the setting component will know that.